#
# mobile_base_controller:
#   type        : "diff_drive_controller/DiffDriveController"
#   rear_wheel: 'rearwheelsjoints'
#   front_steer: 'Frontrightwheeljoints' #, 'Fronleftwheel']
#   publish_rate: 50.0               # default: 50
#   pose_covariance_diagonal : [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]
#   twist_covariance_diagonal: [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]
#
#   # Wheel separation between the rear and the front, and diameter of the rear.
#   # These are both optional.
#   # steer_drive_controller will attempt to read either one or both from the
#   # URDF if not specified as a parameter.
#   wheel_separation_h : 0.5
#   wheel_radius : 0.2
#
#   # Wheel separation and radius multipliers for odometry calibration.
#   wheel_separation_h_multiplier: 1.0 # default: 1.0
#   wheel_radius_multiplier    : 1.0 # default: 1.0
#
#   # Steer position angle multipliers for fine tuning.
#   steer_pos_multiplier       : 1.0
#
#   # Velocity commands timeout [s], default 0.5
#   cmd_vel_timeout: 0.25
#
#   # Base frame_id
#   base_frame_id: chassis #default: base_link
#
#   # Odom frame_id
#   odom_frame_id: odom
#
#   # Velocity and acceleration limits
#   # Whenever a min_* is unspecified, default to -max_*
#   linear:
#     x:
#       has_velocity_limits    : true
#       max_velocity           : 1.0  # m/s
#       min_velocity           : -0.5 # m/s
#       has_acceleration_limits: true
#       max_acceleration       : 0.8  # m/s^2
#       min_acceleration       : -0.4 # m/s^2
#       has_jerk_limits        : true
#       max_jerk               : 5.0 # m/s^3
#
#   angular:
#     z:
#       has_velocity_limits    : true
#       max_velocity           : 1.7  # rad/s
#       has_acceleration_limits: true
#       max_acceleration       : 1.5  # rad/s^2
#       has_jerk_limits        : true
#       max_jerk               : 2.5 # rad/s^3
#
#   gains:
#     rearwheelsjoints  :  {p: 100000.0, d: 10.0, i: 0.50, i_clamp: 3.0}


toycar:
  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50

  # Position Controllers ---------------------------------------
  leftjoint_position_controller:
    type: effort_controllers/JointPositionController
    joint: left_frontaxisjoint
    pid: {p: 50.0, i: 0.01, d: 10.0}
  rightjoint_position_controller:
    type: effort_controllers/JointPositionController
    joint: right_frontaxisjoint
    pid: {p: 50.0, i: 0.01, d: 10.0}
  lidar_controller:
    type: velocity_controllers/JointVelocityController
    joint: laser_joint
    pid:
      p: 50.0
      i: 0.01
      d: 10.0

  # Velocity Controllers ---------------------------------------
  leftwheel_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: FrontLeftWheelJoint
    pid:
      p: 100.0
      i: 0.01
      d: 10.0

  rightwheel_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: FrontRightWheelJoint
    pid:
      p: 100.0
      i: 0.01
      d: 10.0
  # velocity_controller:
  #   type        : "diff_drive_controller/DiffDriveController"
  #   left_wheel  : 'FrontLeftWheelJoint'
  #   right_wheel : 'FrontRightWheelJoint'
  #   publish_rate: 50.0               # default: 50
  #   pose_covariance_diagonal : [0.001, 0.001, 0.0, 0.0, 0.0, 0.0]
  #   twist_covariance_diagonal: [0.001, 0.001, 0.0, 0.0, 0.0, 0.0]

  # # Wheel separation and diameter. These are both optional.
  # # diff_drive_controller will attempt to read either one or both from the
  # # URDF if not specified as a parameter
  #   wheel_separation : 0.4318
  #   wheel_radius : 0.1016
  #
  # # Wheel separation and radius multipliers for odometry calibration.
  #   wheel_separation_h_multiplier: 2.5 # default: 1.0
  #   wheel_radius_multiplier    : 1.0 # default: 1.0
  #
  # # Steer position angle multipliers for fine tuning.
  #   steer_pos_multiplier       : 1.0
  #
  #
  # # Base frame_id
  #   base_frame_id: base_link #default: base_link




# # Velocity Controllers ---------------------------------------
#   leftjoint_velocity_controller:
#     type: JointVelocityController
#     joint: Fronleftwheel
#     pid:
#       p: 100
#       i: 0.01
#       d: 10
#
#   rightjoint_velocity_controller:
#     type: JointVelocityController
#     joint: Frontrightwheeljoints
#     pid:
#       p: 100
#       i: 0.01
#       d: 10

### Trial
# position_trajectory_controller:
#   type: "position_controllers/JointTrajectoryController"
#   joints:
#     - Frontrightwheeljoints
#     - Fronleftwheel

# velocity_trajectory_controller:
#   type: "velocity_controllers/JointTrajectoryController"
#   joints:
#     - Frontrightwheeljoints
#     - Fronleftwheel
#
#   gains:
#     Frontrightwheeljoints : {p: 100, d: 1, i: 1}
#     Fronleftwheel : {p: 100, d: 1, i: 1}
#
#   velocity_ff:
#         Frontrightwheeljoints : 1
#         Fronleftwheel : 1
